cdap-unit-test/src/main/java/co/cask/cdap/test/RevealingTxSystemClient.java

```diff
@@ -0,0 +1,130 @@
+import org.apache.tephra.TransactionNotInProgressException;
+import org.apache.tephra.TransactionSystemClient;
+import org.apache.tephra.inmemory.InMemoryTxSystemClient;
+
+import java.io.InputStream;
+import java.util.Collection;
+import java.util.Set;
+
+/**
+ * A TransactionSystemClient that creates transaction objects with additional fields for validation.
+ */
+public class RevealingTxSystemClient implements TransactionSystemClient {
+
+  private final InMemoryTxSystemClient txClient;
+
+  @Inject
+  RevealingTxSystemClient(InMemoryTxSystemClient txClient) {
+    this.txClient = txClient;
+  }
+
+  @Override
+  public Transaction startLong() {
+    return txClient.startLong();
+  }
+
+  @Override
+  public Transaction startShort() {
+    return txClient.startShort();
+  }
+
+  @Override
+  public Transaction startShort(int timeout) {
+    return new RevealingTransaction(txClient.startShort(timeout), timeout);
+  }
+
+  @Override
+  public boolean canCommit(Transaction tx, Collection<byte[]> changeIds) throws TransactionNotInProgressException {
+    return txClient.canCommit(tx, changeIds);
+  }
+
+  @Override
+  public boolean commit(Transaction tx) throws TransactionNotInProgressException {
+    return txClient.commit(tx);
+  }
+
+  @Override
+  public void abort(Transaction tx) {
+    txClient.abort(tx);
+  }
+
+  @Override
+  public boolean invalidate(long tx) {
+    return txClient.invalidate(tx);
+  }
+
+  @Override
+  public Transaction checkpoint(Transaction tx) throws TransactionNotInProgressException {
+    return txClient.checkpoint(tx);
+  }
+
+  @Override
+  public InputStream getSnapshotInputStream() throws TransactionCouldNotTakeSnapshotException {
+    return txClient.getSnapshotInputStream();
+  }
+
+  @Override
+  public String status() {
+    return txClient.status();
+  }
+
+  @Override
+  public void resetState() {
+    txClient.resetState();
+  }
+
+  @Override
+  public boolean truncateInvalidTx(Set<Long> invalidTxIds) {
+    return txClient.truncateInvalidTx(invalidTxIds);
+  }
+
+  @Override
+  public boolean truncateInvalidTxBefore(long time) throws InvalidTruncateTimeException {
+    return txClient.truncateInvalidTxBefore(time);
+  }
+
+  @Override
+  public int getInvalidSize() {
+    return txClient.getInvalidSize();
+  }
+
+  /**
+   * This transaction class allows us to return additional details about the transaction to the client.
+   * For now, it is only the transaction timeout, but we can add more information later.
+   * This can then be used by test cases to validate that the transaction was started with the right properties.
+   */
+  public static class RevealingTransaction extends Transaction {
+
+    // making this field public: Because this will be loaded in a different classloader than the test app,
+    // the app cannot cast to this class and call a getter. Since it will have to use reflection anyway, it may
+    // as well access the field directly.
```

Comment:
Since this is a test class that we are introducing solely for the test case, I think it is ok to use reflection to validate it. It makes it clear in the test case code that we rely on the helper class to have that field. 

If we change the toString() method, then someone who reads the code will look at Transaction.toString() and wonder where the timeout= comes from.

I really don't know what's better and how important it is.